ROX-29263: Update docs for RH vulns VEX (#94339)
openshift/openshift-docs: modules/con-vuln-sources.adoc

Scanner V4 uses the following vulnerability sources:

> link:https://security.access.redhat.com/data/csaf/v2/vex/[Red{nbsp}Hat VEX]:: This source is used with release 4.6 and later. This source provides vulnerability data in link:https://docs.oasis-open.org/csaf/csaf/v2.0/os/csaf-v2.0-os.html#45-profile-5-vex[Vulnerability Exploitability eXchange(VEX)] format. {product-title-short} takes advantage of VEX benefits to significantly decrease the time needed for the initial loading of vulnerability data, and the space needed to store vulnerability data.

Reviewer: If we don't mention its improved accuracy over OVAL anywhere else, we should do that here, too

+
{product-title-short} might list a different number of vulnerabilities when you are scanning with a {product-title-short} version that uses OVAL, such as {product-title-short} version 4.5, and a version that uses VEX, such as version 4.6. For example, {product-title-short} no longer displays vulnerabilities with a status of "under investigation," while these vulnerabilities were included with previous versions that used OVAL data.
+

> For containers from the Red Hat Ecosystem Catalog, {product-title-short} provides an option to show only vulnerabilities from VEX data. VEX data for Red Hat images is the most accurate because vulnerabilities in those images have been vetted by the Red Hat security team and the results reported in VEX. Other vulnerability sources such as OSV can report vulnerabilities that Red Hat has determined are not applicable to the image. This can cause false positives in vulnerability results. Red Hat also uses VEX to provide link:https://access.redhat.com/security/middleware_security_scanning_problem[accurate security data for Middleware products].

Reviewer 2: Supplemental style guide: [Use a non-breaking space ({nbsp}) between the words "Red" and "Hat".](https://redhat-documentation.github.io/supplementary-style-guide/#non-breaking-spaces)

```suggestion
For containers from the Red Hat Ecosystem Catalog, {product-title-short} provides an option to show only vulnerabilities from VEX data. VEX data for Red{nbsp}Hat images is the most accurate because vulnerabilities in those images have been vetted by the Red{nbsp}Hat security team and the results reported in VEX. Other vulnerability sources such as OSV can report vulnerabilities that Red Hat has determined are not applicable to the image. This can cause false positives in vulnerability results. Red{nbsp}Hat also uses VEX to provide link:https://access.redhat.com/security/middleware_security_scanning_problem[accurate security data for Middleware products].
```

> For containers from the Red Hat Ecosystem Catalog, {product-title-short} provides an option to show only vulnerabilities from VEX data. VEX data for Red Hat images is the most accurate because vulnerabilities in those images have been vetted by the Red Hat security team and the results reported in VEX. Other vulnerability sources such as OSV can report vulnerabilities that Red Hat has determined are not applicable to the image. This can cause false positives in vulnerability results. Red Hat also uses VEX to provide link:https://access.redhat.com/security/middleware_security_scanning_problem[accurate security data for Middleware products].

Reviewer: ```suggestion
For containers from or built on top of container from the Red Hat Ecosystem Catalog, {product-title-short} provides an option to show only vulnerabilities from Red Hat's VEX data. VEX data for Red Hat images is the most accurate because vulnerabilities in those images have been vetted by the Red Hat security team. Other vulnerability sources such as OSV can report vulnerabilities that Red Hat has determined are not applicable to the image. This can cause false positives in vulnerability results. Enabling the setting to only use VEX data for Red Hat images will minimize these false positives.
```

+

> The option to use VEX data for Red Hat containers is disabled by default. To enable this option, in Scanner V4 Matcher, set the environment variable `ROX_SCANNER_V4_RED_HAT_LAYERS_RED_HAT_VULNS_ONLY` to `true`. Be aware that in rare instances, using this option can cause valid vulnerabilities to not appear in scan results. For example, Red Hat does not track vulnerabilities for products that have reached end of life.

Reviewer 2: ```suggestion
The option to use VEX data for Red{nbsp}Hat containers is disabled by default. To enable this option, in Scanner V4 Matcher, set the environment variable `ROX_SCANNER_V4_RED_HAT_LAYERS_RED_HAT_VULNS_ONLY` to `true`. Be aware that in rare instances, using this option can cause valid vulnerabilities to not appear in scan results. For example, Red{nbsp}Hat does not track vulnerabilities for products that have reached end of life.
```

Reviewer 2: > Scanner V4 Matcher

Do folks know how to set envvars for it?

> The option to use VEX data for Red Hat containers is disabled by default. To enable this option, in Scanner V4 Matcher, set the environment variable `ROX_SCANNER_V4_RED_HAT_LAYERS_RED_HAT_VULNS_ONLY` to `true`. Be aware that in rare instances, using this option can cause valid vulnerabilities to not appear in scan results. For example, Red Hat does not track vulnerabilities for products that have reached end of life.

Reviewer: ```suggestion
The option to use VEX data for Red Hat containers is disabled by default. To enable this option, in Scanner V4 Matcher, set the environment variable `ROX_SCANNER_V4_RED_HAT_LAYERS_RED_HAT_VULNS_ONLY` to `true`. Be aware that in rare instances, using this option can cause valid vulnerabilities to not appear in scan results (false negatives). For example, Red Hat does not track vulnerabilities for products that have reached end of life. Also be aware Red Hat's VEX data is missing link:https://access.redhat.com/security/middleware_security_scanning_problem[accurate security data for many Middleware products].
```

Author: Gotcha - FWIW, that page is confusing and I interpreted it totally differently!

> For containers from or built on top of containers from the Red{nbsp}Hat Ecosystem Catalog, {product-title-short} provides an option to show only vulnerabilities from Red{nbsp}Hat's VEX data. VEX data for Red{nbsp}Hat images is the most accurate because vulnerabilities in those images have been vetted by the Red{nbsp}Hat security team and the results reported in VEX. Other vulnerability sources such as OSV can report vulnerabilities that Red{nbsp}Hat has determined are not applicable to the image. This can cause false positives in vulnerability results. Enabling the setting to use only VEX data for Red{nbsp}Hat images minimizes these false positives.

Reviewer: Looks good, but I am confused by the meaning of

> and the results reported in VEX

Can you help me understand what this is trying to say?

Author: Clumsily trying to say that VEX is most accurate because Red Hat security vets the images and reports vulnerability results in VEX. I'll try to rephrase.

+
{product-title-short} might list a different number of vulnerabilities when you are scanning with a {product-title-short} version that uses OVAL, such as {product-title-short} version 4.5, and a version that uses VEX, such as version 4.6. For example, {product-title-short} no longer displays vulnerabilities with a status of "under investigation," while these vulnerabilities were included with previous versions that used OVAL data.
+

> For containers from the Red Hat Ecosystem Catalog, {product-title-short} provides an option to show only vulnerabilities from VEX data. VEX data for Red Hat images is the most accurate because vulnerabilities in those images have been vetted by the Red Hat security team and the results reported in VEX. Other vulnerability sources such as OSV can report vulnerabilities that Red Hat has determined are not applicable to the image. This can cause false positives in vulnerability results. The option to use VEX data for Red Hat containers is disabled by default. To enable this option, in Scanner V4 Matcher, set the environment variable `ROX_SCANNER_V4_RED_HAT_LAYERS_RED_HAT_VULNS_ONLY` to `true`. Be aware that in rare instances, using this option can cause valid vulnerabilities to not appear in scan results. For example, Red Hat does not track vulnerabilities for products that have reached end of life.

Reviewer: Not sure if we want to call out Middleware, too - https://access.redhat.com/security/middleware_security_scanning_problem

+
For more information about Red Hat security data, including information about the use of OVAL, Common Security Advisory Framework Version 2.0 (CSAF), and VEX, see link:https://www.redhat.com/en/blog/future-red-hat-security-data[The future of Red Hat security data].

> https://access.redhat.com/security/data/metrics/cvemap.xml[Red{nbsp}Hat CVE Map]:: This is used in addition with VEX data for images which appear in the link:https://catalog.redhat.com/software/containers/explore[Red{nbsp}Hat Ecosystem Catalog].

Reviewer: Mentioned in a different PR, but as of 4.7 we don't use this anymore. Forgot to mention this last release... Sorry about that!
